jenkins_build.sh

```diff
@@ -1,36 +1,65 @@
-set -e
+#!/bin/bash
+set -ex
+
+# Required parameters
+# REPO_NAME: the name of the repo this job will checkout
+# BRANCH: the branch of the repo this job will use
+# ZENDEV_REPO: the zendev repo
+# ZENDEV_BRANCH: the zendev branch this job will use
+# ZENDEV_VER: the zendev version
+# GO_VER: the go version
+
+check_var() {
+    if [ -z "$1" ]; then
+        echo Undefined variable
+        exit 1
+    fi
+}
 
 REPO_NAME=zenoss-prodbin
-REPO_PATH=$WORKSPACE/${ZENDEV_ENV}/src/github.com/zenoss/${REPO_NAME}
 BRANCH=develop
-# The name of the zendev environment that will be created. This name will be used as the devimg tag as well.
-ZENDEV_ENV=${REPO_NAME}-${JOB_BASE_NAME}
 ZENDEV_REPO=git@github.com:zenoss/zendev.git
-ZENDEV_DIR=zendev
 ZENDEV_BRANCH=zendev2
 ZENDEV_VER=0.2.0
-VENV_DIR=venv
 GO_VER=1.7.4
 
+# The name of the zendev environment that will be created.
+# This name will be used as the devimg tag as well.
+ZENDEV_ENV=${REPO_NAME}-${JOB_BASE_NAME}
+
+# The repo that is checked out will be copied to this path
+# inside zendev.
+REPO_PATH=$WORKSPACE/${ZENDEV_ENV}/src/github.com/zenoss/${REPO_NAME}
+
+# Check if all required parameters are defined.
+echo REPO_NAME;check_var $REPO_NAME;echo OK
+echo BRANCH;check_var $BRANCH;echo OK
+echo ZENDEV_REPO;check_var $ZENDEV_REPO;echo OK
+echo ZENDEV_BRANCH;check_var $ZENDEV_BRANCH;echo OK
+echo ZENDEV_VER;check_var $ZENDEV_VER;echo OK
+echo GO_VER;check_var $GO_VER;echo OK
+
 echo Creating a virtual env...
-virtualenv ${VENV_DIR}
-source ${VENV_DIR}/bin/activate
-git clone ${ZENDEV_REPO} ${ZENDEV_DIR}
-cd ${ZENDEV_DIR};git checkout ${ZENDEV_BRANCH}
+virtualenv venv
+source venv/bin/activate
+git clone ${ZENDEV_REPO} zendev
+cd zendev;git checkout ${ZENDEV_BRANCH}
 pip install -e .
 zendev version | grep ${ZENDEV_VER} # Check zendev version
```

Comment:
i think you need to change this line so that the echo on the next line will report the version we ARE using as opposed to version we expect to use. Better yet, check the actual version against the expected version and exit if they are not the same
```
ZENDEV_VER_ACTUAL=`zendev version`
if [ "${ZENDEV_VER_ACTUAL}" != "${ZENDEV_VER}" ]; then
    echo "ERROR: expected zendev version ${ZENDEV_VER}, but found zendev version ${ZENDEV_VER_ACTUAL} "
    exit 1
fi
```